### PR TITLE
feature: add blockstream.info current block height

### PR DESCRIPTION
**Pull request by reez**
## Description

Added the blockstream.info request for block height. 

Made sure I covered testnet and mainnet urls:
- https://blockstream.info/testnet/api/blocks/tip/height
- https://blockstream.info/api/blocks/tip/height

Added a `fetch` method because the blockstream height url doesn't give a similar json response to the other height urls.  I tried to think of the best and most simple way to accomplish this, and it was just much cleaner to add an overload method that was in the same style as the existing `fetch` method. Happy to take any feedback if there is a cooler way to do it.

Added a `PinnedHost` for blockstream.info, and also added the 2 certs to the project assets. Let me know if I missed anything on that part.

Also, made `BlockchainHeight` conform to `CaseIterable`.

## Motivation and Context

Issue [53](https://github.com/LN-Zap/zap-iOS/issues/53)

## How Has This Been Tested?

Ran unit tests and they all passed.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: Lightning/Services/Misc/BlockChainHeightUpdater.swift [hunk 1]
```diff
@@ -70,7 +72,24 @@ enum BlockchainHeight {
...
+            }
+        }
+    }
+    
+    private func fetch(url: URL, completion: @escaping (Int) -> Void, map: @escaping (String) -> Int?) {
+        let task = URLSession.pinned.dataTask(with: url) { data, _, error in
+            if error != nil {
+                print(String(describing: error))
+            } else if let data = data,
+                let dictionary = String(data: data, encoding: .utf8),
```
Comment (ottosuess) [inline Lightning/Services/Misc/BlockChainHeightUpdater.swift]: might be a bit confusing that the string is called dictionary 😉

File: Lightning/Services/Misc/BlockChainHeightUpdater.swift [hunk 2]
```diff
@@ -70,7 +72,24 @@ enum BlockchainHeight {
...
                 $0["height"] as? Int
             }
+        case .blockstream:
+            fetch(url: url(for: network), completion: completion) {
+                Int($0)
+            }
+        }
+    }
+    
+    private func fetch(url: URL, completion: @escaping (Int) -> Void, map: @escaping (String) -> Int?) {
```
Comment (ottosuess) [inline Lightning/Services/Misc/BlockChainHeightUpdater.swift]: i'd remove the map argument. just to keep things simple. and just cast the string to an `Int` later in the method.

File: Lightning/Services/Misc/BlockChainHeightUpdater.swift (hunk 1 above)
Comment (reez) [reply Lightning/Services/Misc/BlockChainHeightUpdater.swift]: I was trying to stick so closely to the other fetch method in shape and naming, that I actually went way too far ha (naming a string `dictionary` is blatantly confusing), so good call out, thanks, will change!

File: Lightning/Services/Misc/BlockChainHeightUpdater.swift (hunk 2 above)
Comment (reez) [reply Lightning/Services/Misc/BlockChainHeightUpdater.swift]: Is this what you're thinking?:

```
    private func fetch(url: URL, completion: @escaping (Int) -> Void) {
        let task = URLSession.pinned.dataTask(with: url) { data, _, error in
            if error != nil {
                print(String(describing: error))
            } else if let data = data,
                let string = String(data: data, encoding: .utf8),
                let height = Int(string) {
                completion(height)
            }
        }
        task.resume()
    }
```

Comment (ottosuess) [reply Lightning/Services/Misc/BlockChainHeightUpdater.swift]: yes! 😍

**Comment by reez** [comment]
Updated the PR, thanks for the solid feedback! 😎

**Pull request merged**